### PR TITLE
formatter_ltsv: suppress delimiters in output

### PR DESCRIPTION
**Pull request by MikePlayle**
Replace tabs with spaces before writing them to the output, to prevent
creating malformed LTSV files.

Also add test cases to verify this behaviour.

### CODE REVIEW

File: lib/fluent/plugin/formatter_ltsv.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def format(tag, time, record)
         formatted = ""
         record.each do |label, value|
           formatted << @delimiter if formatted.length.nonzero?
-          formatted << "#{label}#{@label_delimiter}#{value}"
+          formatted << "#{label}#{@label_delimiter}#{value.to_s().gsub(@delimiter, ' ')}"
```
Comment (repeatedly) [inline lib/fluent/plugin/formatter_ltsv.rb]: Replaced char should be configurable, not only `' '`.

Comment (repeatedly) [inline lib/fluent/plugin/formatter_ltsv.rb]: And `to_s` is better than `to_s()`

**Comment by kenhys** [comment]
@MikePlayle do you have still motivation to improve this PR?

**Comment by MikePlayle** [comment]
I forgot about this completely. I thought it had been merged! I'll get to it this weekend.

**Comment by ashie** [comment]
Although I felt a doubt a bit about the specification, probably it's a reasonable solution when I consider it based on philosophy of LTSV (parser should be as simple as possible). I'll merge it after checking the tests results on my environment.

**Review by ashie — APPROVED** [review]

**Comment by ashie** [comment]
Sorry I wrongly pushed master branch.
I'll open new PR with revised patch.

**Pull request closed without merging**